### DISCUSSION TITLE
How to implement ReloadSchemaIndicator to change schema in runtime?

### DISCUSSION THREAD

**Discussion by oguztozkoparan** [Netflix/dgs-framework#1019] [Q&A] [answered]
I have a Spring Boot backend already DGS implemented with changing database: users can create new tables and new columns, also, can create relations. What I want is to create some dynamic structure which does build new queries according to created tables and columns. I read about the ReloadSchemaIndicator and Dynamic Schemas documentation. But I could not figure it out how can I do it. Maybe it is impossible to change graphql schema in runtime. Do anyone know? Also, I think we need more information about ReloadSchemaIndicator on documentation because I could not understand if I did not read the source code.

**Comment by paulbakker** [comment, marked as answer]
The `ReloadSchemaIndicator` is really just logic that gives `true` or `false`. By default, it will always use the cached schema for obvious performance reasons. You implement a `ReloadSchemaIndicator` with whatever logic you need to decide "I need to build a fresh schema".
At Netflix we detect when we run on a developer machine (as opposed to the cloud), and _always_ reload the schema in that case.

You could maybe trigger an event based on the user making a change to the database and set it to reload.

**Reply by oguztozkoparan** [reply]
Thank you very much for your answer, I was able to understand thanks to you.
